### ISSUE TITLE
ListView Headers do not Stick inside ScrollView

### ISSUE THREAD

**Issue by joncursi** [facebook/react-native#10629]
The desired interaction is: 

1) a list with sticky section headers
2) pull down at the top to refresh

When a ListView with sticky headers is placed within a ScrollView, the headers lose their "stick":

```jsx
<ScrollView
  refreshControl={
    <RefreshControl
      ...
    />
>
  <ListView
    dataSource={...}
    renderRow={...}
    renderSectionHeader={...} // these renderables no longer stick!
  />
</ScrollView>
```

This was also called out on a [separate issue a while ago](https://github.com/facebook/react-native/issues/1974#issuecomment-144799173), but it looks like an issue was never created. So here we go!

**Comment by brentvatne** [comment]
You could do this instead:

```js
<ListView
  dataSource={...}
  renderRow={...}
  renderSectionHeader={...} // these renderables no longer stick!
  renderScrollComponent={props => {
    return (
      <ScrollView {...props}
        refreshControl={
          <RefreshControl
            ...
          />
        }
      />
    );
  }}
/>
```

Generally speaking, you should try to avoid nesting ScrollViews when possible (and ListView renders a ScrollView).

**Comment by joncursi** [comment]
@brentvatne thanks for your reply. I tried out your proposed solution, however the result was the same as mentioned above when I explicitly wrapped the `ListView` within a `ScrollView`.

In your proposed example, `RefreshControl` does indeed get passed to ScrollView via `renderScrollComponent`, however when we do this, the headers within the `ListView` continue to lose their "stickyness".

Perhaps we need to add a `refreshControl` prop directly to `ListView` itself to make this work. Can re-open this thread to improve visibility on this issue? Thanks!

**Comment by brentvatne** [comment]
@joncursi - I can't repro this, it actually works fine for me to set `refreshControl` directly on the `ListView`

Code: https://gist.github.com/brentvatne/014e773b3824cdcb67e3dbcee24e70be#file-refreshcontrol-js-L30-L49
Try it here: https://getexponent.com/@community/list-view-refreshcontrol-test

Happy to reopen if there is a repro case

**Comment by joncursi** [comment]
Ah, I did not realize that ListView has access to all of the props available on ScrollView. I see that now in the documentation. I just tried setting `refreshControl` directly on ListView and was able to achieve the desired interaction. Thank you @brentvatne !

**Comment by nirmal3win87** [comment]
Is there any solution for sticky header not working inside scroll view?

**Comment by palexs** [comment]
Hi, @brentvatne

In our app we need to achieve double sticky headers. So there should be 1 common sticky header for entire table and sticky headers for each `SectionList` section.
But whenever I wrap `SectionList` inside container `ScrollView` (or other `SectionList`) its sections headers loose their stickiness. Is there a way to achieve the desired result?

![double-sticky-headers](https://user-images.githubusercontent.com/1583824/30422315-3e00bbfe-9948-11e7-87ce-b070d1f61f86.gif)

So, basically I somehow need to make those "Section Title #" stick under that red "Column Sticky Header". Current implementation wraps `SectionList` inside parent `SectionsList`.
